backend/backend/settings/common.py

```diff
@@ -95,10 +95,16 @@ def to_bool(value):
     'django.contrib.auth.middleware.RemoteUserMiddleware',
     'django.contrib.messages.middleware.MessageMiddleware',
     'django.middleware.clickjacking.XFrameOptionsMiddleware',
-    'libs.sql_printing_middleware.SQLPrintingMiddleware',
     'libs.health_check_middleware.HealthCheckMiddleware',
 ]
 
+
+DJANGO_LOG_SQL_QUERIES = to_bool(os.environ.get('DJANGO_LOG_SQL_QUERIES', 'True'))
```

Comment:
should be `to_bool(os.environ.get('DJANGO_LOG_SQL_QUERIES', True))`

Reply:
Yes, I should have done that to keep a consistent code base.

My bad, I've been used to define a string as a default value of the `os.environ.get()` function. This way if the env var exists or not, the return value is always a string.

Reply:
Why should it be always a string?

Reply:
It's not compulsory, it's a rule I've been used to follow. It allows to have the same behaviour if the env var is defined.

For instance the following condition will be true if the env var 'ENABLE_FEATURE' is not defined:
```python
str(os.environ.get('ENABLE_FEATURE', False))
```
It's pretty obvious that above code is wrong, but it could be misleading.

Returning always a string allows also to do this kind of operation:
```python
myvar = os.environ.get('ENABLE_FEATURE', 'False').lower()
```

Reply:
I don't see your point.
Why would we get a `'False'` or  `'false'` value?

Reply:
This would be the case if the env var is set to false for instance:
```
bash: export VALUE=false
python: assert os.environ.get('VALUE') == 'false'
```

Reply:
But aren't we dealing with Boolean variables here?

Reply:
My point is to ensure that the function `os.environ.get()` always return a string (if the env var exists or if the env var does not exist). Then we can use a function converting a string to the type you want. It allows to deal with a single type in the conversion function.

Reply:
Using `'True'` instead of `True` looks a lot like an error.
Also we have to consider `'True'` is a string and allocate python memory.
`True` is a language keyword.

IMHO `True` is the best default value to use.

Reply:
Close conversation.